tests/basic/sharedv4_svc_test.go

```diff
@@ -259,84 +266,303 @@ var _ = Describe("{Shared4SvcFailoverIO}", func() {
+	var testrailID = 54776
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/cases/view/54776
+	var runID int
+	var contexts, testSv4Contexts []*scheduler.Context
+	var workers []node.Node
+
+	JustBeforeEach(func() {
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+
+	It("has to verify that the server unexports and re-exports volume to the client node "+
+		"after pod goes away and comes back", func() {
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("client-restart-%d", i))...)
+		}
+		testSv4Contexts = getTestSv4Contexts(contexts)
+		if len(testSv4Contexts) == 0 {
+			Skip("No test-sv4-svc apps were found")
+		}
+		workers = node.GetWorkerNodes()
+
+		Step("scale the test-sv4-svc apps so that one pod runs on each worker node", func() {
+			scaleApps(testSv4Contexts, len(workers))
+		})
+		ValidateApplications(contexts)
+
+		for _, ctx := range testSv4Contexts {
+			Step(fmt.Sprintf("scale the deployment down and then up for app %s", ctx.App.Key), func() {
+				var attachedNode *node.Node
+				var nodeWithNoPod *node.Node
+
+				vols, err := Inst().S.GetVolumes(ctx)
+				Expect(err).NotTo(HaveOccurred())
+
+				apiVol, err := Inst().V.InspectVolume(vols[0].ID)
+				Expect(err).NotTo(HaveOccurred())
+
+				attachedNode, err = Inst().V.GetNodeForVolume(vols[0], defaultCommandTimeout, defaultCommandRetry)
+				Expect(err).NotTo(HaveOccurred())
+				logrus.Infof("volume %v (%v) is attached to node %v", vols[0].ID, apiVol.Id, attachedNode.Name)
+
+				exportsBeforeScaleDown := getExportsOnNode(apiVol, attachedNode)
+
+				// Scale down the app by 1 and check if the pod on NFS client node got removed.
+				// If not, scale down by 1 more. It should not take more than 2 attempts since
+				// there is no more than 1 pod running on each node.
+			Outer:
+				for i := 1; i < 3; i++ {
+					scaleApp(ctx, len(workers)-i)
+					ValidateContext(ctx)
+
+					pods, err := core.Instance().GetPodsUsingPV(vols[0].ID)
+					Expect(err).NotTo(HaveOccurred())
+
+					podsByWorker := map[string]corev1.Pod{}
+					for _, pod := range pods {
+						podsByWorker[pod.Spec.NodeName] = pod
+					}
+					for _, worker := range workers {
+						if _, ok := podsByWorker[worker.Name]; !ok && worker.Name != attachedNode.Name {
+							nodeWithNoPod = &worker
+							break Outer
+						}
+					}
+				}
+				Expect(nodeWithNoPod).NotTo(BeNil(), "did not find NFS client node whose pod was removed")
+
+				// We don't expect a failover
+				validateAttachedNode(vols[0], attachedNode)
+				exportsAfterScaleDown := getExportsOnNode(apiVol, attachedNode)
+
+				Expect(exportsBeforeScaleDown).Should(ContainElement(nodeWithNoPod.DataIp),
+					"client IP not found in the exports before scaling down the app")
+
+				Expect(exportsAfterScaleDown).ShouldNot(ContainElement(nodeWithNoPod.DataIp),
```

Comment:
Lets check if the IPs for the other nodes exist where the pods are still running.

Reply:
Good idea. Added.